packages/engine/Source/Scene/createElevationBandMaterial.js

```diff
@@ -462,46 +462,42 @@ function createLayeredEntries(layers) {
  * });
  */
 function createElevationBandMaterial(options) {
-  options = defaultValue(options, defaultValue.EMPTY_OBJECT);
-  const scene = options.scene;
-  const layers = options.layers;
+  const { scene, layers } = defaultValue(options, defaultValue.EMPTY_OBJECT);
 
   //>>includeStart('debug', pragmas.debug);
   Check.typeOf.object("options.scene", scene);
   Check.defined("options.layers", layers);
   Check.typeOf.number.greaterThan("options.layers.length", layers.length, 0);
   //>>includeEnd('debug');
 
+  const { context } = scene;
   const entries = createLayeredEntries(layers);
   const entriesLength = entries.length;
-  let i;
 
   let heightTexBuffer;
   let heightTexDatatype;
   let heightTexFormat;
 
-  const isPackedHeight = !createElevationBandMaterial._useFloatTexture(
-    scene.context
-  );
+  const isPackedHeight = !createElevationBandMaterial._useFloatTexture(context);
   if (isPackedHeight) {
     heightTexDatatype = PixelDatatype.UNSIGNED_BYTE;
     heightTexFormat = PixelFormat.RGBA;
     heightTexBuffer = new Uint8Array(entriesLength * 4);
-    for (i = 0; i < entriesLength; i++) {
+    for (let i = 0; i < entriesLength; i++) {
       Cartesian4.packFloat(entries[i].height, scratchPackedFloat);
       Cartesian4.pack(scratchPackedFloat, heightTexBuffer, i * 4);
     }
   } else {
     heightTexDatatype = PixelDatatype.FLOAT;
-    heightTexFormat = PixelFormat.LUMINANCE;
+    heightTexFormat = context.webgl2 ? PixelFormat.RED : PixelFormat.LUMINANCE;
```

Comment:
Does `Megatexture.js` need similar treatment [here](https://github.com/CesiumGS/cesium/blob/main/packages/engine/Source/Scene/Megatexture.js#L54-L68)? It's the only other place I noticed where `LUMINANCE` and `LUMINANCE_ALPHA` are being used with `FLOAT` datatype.

Reply:
Thanks @lilleyse! Yes, it did need a similar fix:
```javascript
  if (channelCount === 1) {
    pixelFormat = context.webgl2 ? PixelFormat.RED : PixelFormat.LUMINANCE;
  } else if (channelCount === 2) {
    pixelFormat = context.webgl2 ? PixelFormat.RG : PixelFormat.LUMINANCE_ALPHA;
```